src/dal_select2/static/autocomplete_light/select2.js

```diff
@@ -35,13 +35,20 @@
                     dstName = srcName;
                 }
                 // First look for this field in the inline
-                $field = $('[name=' + prefix + srcName + ']');
+                $field_selector = '[name=' + prefix + srcName + ']';
+                $field = $($field_selector);
                 if (!$field.length)
-                    // As a fallback, look for it outside the inline
-                    $field = $('[name=' + srcName + ']');
-                if ($field.length)
-                    forwardedData[dstName] = $field.val();
-
+                // As a fallback, look for it outside the inline
+                    $field_selector = '[name=' + srcName + ']';
+                $field = $($field_selector);
+                if ($field.length) {
+                    if ($field.attr('type') === 'checkbox')
+                        forwardedData[dstName] = !!$($field_selector + ":checked").val();
+                    else if ($field.attr('type') === 'radio')
+                        forwardedData[dstName] = $($field_selector + ":checked").val();
```

Comment:
Have you tried [$.serializeArray](http://api.jquery.com/serializeArray/) ? It makes a json object with form data, perhaps this would avoid having to maintain "form to json" logic in this script ?